Serverless MCP server initial commit

bx9900 opened the pull request:
## Summary
Creates a MCP server to assist in deploying application onto AWS Serverless

### Changes

Adds tools to interact with Serverless services, including Lambda, IAM, API GW, and DDB. Also adds tools to that provide guidance on how to deploy onto AWS Severless and tools that wrap SAM CLI, fetch example template from Serverless, Lambda Event Schemas.

### User experience

> Please share what the user experience looks like before and after this change

## Checklist

If your change doesn't seem to apply, please leave them unchecked.

* [ ] I have reviewed the [contributing guidelines](https://github.com/awslabs/mcp/blob/main/CONTRIBUTING.md)
* [ ] I have performed a self-review of this change
* [ ] Changes have been tested
* [ ] Changes are documented

Is this a breaking change? (Y/N) N

**RFC issue number**:

Checklist:

* [ ] Migration process documented
* [ ] Implement warnings (if it can live side by side)

## Acknowledgment

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the [project license](https://github.com/awslabs/mcp/blob/main/LICENSE).

bx9900 committed:
Serverless MCP server initial commit

awslabs-mcp committed:
Automatic update of packages

tomron-aws committed:
feat: Kendra Index MCP Server Pull Request (#243)

* Initial commit for Kendra Index MCP server

* updated readme for Kendra Index mcp server

* Fixed pre-commit hook errors

* Fixed errors in test suite for kendra mcp server

* Fixing some things, hope it works

* Modified handling of AWS credentials for Kendra Index MCP Server

* Pre-commit edits

* Fixing unbound variable error and changing tool name

* Fixing tests

* Addressed TODO's in /kendra-index-mcpserver-mcps-server/readme

* Fix pre-commit EOF error

* Changed directory name to amazon-kendra-index-mcp-server to match conventions

* Additional files added for directory name change

* Added docker file to kendra mcp server

* Update README.md

* Update README.md

* Update README.md

* Added new tool for listing kendra indices. Modified Kendra Query tool to optionally take an index and a region to query indexes other than the default

* Build error fixes

* Adding docker healtcheck script

* Updated readme

* Merge and removed duplicate entry for KB retrieval server from readme

* Pre-commit fixes

* Update README.md

* Adding test coverage for kendra list indexes tool

* Changed names of tools methods to match tools names, updated test files

* Moved get kendra client method to utils and fixed typo. Added missing type annotations

* Pre-commit fixes

* Adjusted import namespace for utils

* Fixed return type error in utils

* Changed return type for get_kendra_client

* Adding kendra boto3 stubs to pyproject.toml

---------

Co-authored-by: Laith Al-Saadoon <9553966+theagenticguy@users.noreply.github.com>
Co-authored-by: Alain Krok <alkrok@amazon.com>
Co-authored-by: Scott Schreckengaust <scottschreckengaust@users.noreply.github.com>

awslabs-mcp committed:
Automatic update of packages

silver83 committed:
feat: Aurora DSQL MCP Server (#328)

* (feat) Aurora DSQL MCP Server

---------

Co-authored-by: Scott Schreckengaust <scottschreckengaust@users.noreply.github.com>
Co-authored-by: Alain Krok <alkrok@amazon.com>

awslabs-mcp committed:
Automatic update of packages

nitinahuja89 committed:
feat(documentdb): add DocumentDB MCP server (#362)

* feat(documentdb): add DocumentDB MCP server with comprehensive database tools

Implements a new MCP server for AWS DocumentDB that provides tools for connection management, querying, data manipulation, and database administration. Includes the following capabilities:

- Connection Management: connect/disconnect to DocumentDB clusters

- Query Tools: find documents and run aggregation pipelines

- Write Operations: insert, update, and delete documents

- Database Management: list databases, create/list/drop collections

- Analytics: document counting, database/collection stats, schema analysis

- Configurable read-only mode for security constraints

* fix(tests): resolve pyright type errors in documentdb mcp server

* refactor(analytics): improve exception handling in get_field_type function

Replace generic exception handling with specific exception types (ValueError, IndexError, KeyError, TypeError, AttributeError) in the get_field_type helper function. Add warning logs with document details for better error visibility and debugging.

---------

Co-authored-by: Nitin Ahuja <nitahuja@amazon.com>
Co-authored-by: Laith Al-Saadoon <9553966+theagenticguy@users.noreply.github.com>
Co-authored-by: Alain Krok <alkrok@amazon.com>

jimini55 committed:
feat(new MCP server): Add Frontend MCP Server (#241)

* feat(new MCP server):Add Frontend MCP Server for React development

This commit adds a new Frontend MCP Server to provide specialized tools for modern web application development. Key implementations include:

- Three specialized tools:

 - BaseUserInterfaceWebApp for React application setup
 -OptimisticUI for implementing React Query and Zustand patterns
 - UsingAmplifyAuthenticator for AWS Amplify authentication
- Comprehensive test coverage (100%) with appropriate test cases

- Complete documentation in README.md and integration with docs site

- References in top-level README.md and docs/index.md

The server follows MCP design guidelines and provides practical guidance for frontend developers working with modern React applications.

* docs(frontend-mcp-server): Add top-level readme and gitdocs

- Complete documentation in README.md and integration with docs site
- References in top-level README.md and docs/index.md

* tests(frontend-mcp-server): precommit, ruff

* feat(frontend-mcp-server): update GetReactDocsByTopic tool and documentation

- Fix Literal type annotations to include all React documentation topics
- Improve test coverage by properly mocking dependencies
- Add dedicated test_file_utils.py for file loading utilities
- Rename customizing-the-application.md to amplify-theme-customization.md
- Enhance creating-components.md with AWS Amplify focused content
- Update READMEs to accurately reflect current implementation

* tests(frontend-mcp-server): precommit, ruff

* fix(frontend): deleted unrequired static files and topic

* fix(frontend): overwritten static md files

* tests(frontend): cleanup deleted topics

* feat(frontend): update frontend folder repo link

* fix(frontend): fix git folder name

* tests(frontend): precommit run

* Tweaks to improve reliability

* Update index.md

* Update index.md

* tests(frontend): precommit run

* Update README.md

* Update README.md

---------

Co-authored-by: Laith Al-Saadoon <9553966+theagenticguy@users.noreply.github.com>
Co-authored-by: Scott Schreckengaust <scottschreckengaust@users.noreply.github.com>
Co-authored-by: Sanju Sunny <sunsanju@amazon.com>
Co-authored-by: Sanju Sunny <sanjusunny@gmail.com>
Co-authored-by: Alain Krok <alkrok@amazon.com>

seaofawareness committed:
docs: change relative link to absolute for ElastiCache connection ins… (#384)

* docs: change relative link to absolute for ElastiCache connection instructions

* chore: Add codeowners for Amazon ElastiCache valkey and memcached mcp servers

---------

Co-authored-by: Scott Schreckengaust <scottschreckengaust@users.noreply.github.com>

awslabs-mcp committed:
Automatic update of packages

tomron-aws committed:
fix(KendraIndex): missing dependency in pyproject.toml (#387)

* Initial commit for Kendra Index MCP server

* updated readme for Kendra Index mcp server

* Fixed pre-commit hook errors

* Fixed errors in test suite for kendra mcp server

* Fixing some things, hope it works

* Modified handling of AWS credentials for Kendra Index MCP Server

* Pre-commit edits

* Fixing unbound variable error and changing tool name

* Fixing tests

* Addressed TODO's in /kendra-index-mcpserver-mcps-server/readme

* Fix pre-commit EOF error

* Changed directory name to amazon-kendra-index-mcp-server to match conventions

* Additional files added for directory name change

* Added docker file to kendra mcp server

* Update README.md

* Update README.md

* Update README.md

* Added new tool for listing kendra indices. Modified Kendra Query tool to optionally take an index and a region to query indexes other than the default

* Build error fixes

* Adding docker healtcheck script

* Updated readme

* Merge and removed duplicate entry for KB retrieval server from readme

* Pre-commit fixes

* Update README.md

* Adding test coverage for kendra list indexes tool

* Changed names of tools methods to match tools names, updated test files

* Moved get kendra client method to utils and fixed typo. Added missing type annotations

* Pre-commit fixes

* Adjusted import namespace for utils

* Fixed return type error in utils

* Changed return type for get_kendra_client

* Adding kendra boto3 stubs to pyproject.toml

* Adding additional dependencies and fixing type n util import identified during testing

* Precommit fixes:

* Changing utils import namespace in server.py

* Removing unused import from server.py

* Uv lock upversioning

---------

Co-authored-by: Laith Al-Saadoon <9553966+theagenticguy@users.noreply.github.com>
Co-authored-by: Alain Krok <alkrok@amazon.com>
Co-authored-by: Scott Schreckengaust <scottschreckengaust@users.noreply.github.com>

mikegc-aws committed:
feat: MCP Lambda Handler (#373)

* mcp-lambda-handler

* fix

* MCP Lambda Handler Module

* mcp-lambda-handler samples (sam)

* update to point to sample

* awslabs prefix

* checkov fixes (inc remove sample)

* mcp-lambda-handler

* fix

* MCP Lambda Handler Module

* mcp-lambda-handler samples (sam)

* update to point to sample

* awslabs prefix

* checkov fixes (inc remove sample)

* fix: pyproject

Signed-off-by: Scott Schreckengaust <scottschreckengaust@users.noreply.github.com>

* fix: end-of-files and trim whitespace

Signed-off-by: Scott Schreckengaust <scottschreckengaust@users.noreply.github.com>

* fix: ruff

Signed-off-by: Scott Schreckengaust <scottschreckengaust@users.noreply.github.com>

* fix: ruff issues

Signed-off-by: Scott Schreckengaust <scottschreckengaust@users.noreply.github.com>

* fix: python version file

Signed-off-by: Scott Schreckengaust <scottschreckengaust@users.noreply.github.com>

* fix: remove addopts and add pytest dependencies

Signed-off-by: Scott Schreckengaust <scottschreckengaust@users.noreply.github.com>

* fix: update uv.lock

Signed-off-by: Scott Schreckengaust <scottschreckengaust@users.noreply.github.com>

* fix: trailing whitespace and end of file

Signed-off-by: Scott Schreckengaust <scottschreckengaust@users.noreply.github.com>

* fix: ignore false positive pyright

Signed-off-by: Scott Schreckengaust <scottschreckengaust@users.noreply.github.com>

* fix: use only the top-level

Signed-off-by: Scott Schreckengaust <scottschreckengaust@users.noreply.github.com>

* fix: no local pre-commit hooks

Signed-off-by: Scott Schreckengaust <scottschreckengaust@users.noreply.github.com>

* Fix: whitespace and file endings via pre-commit

* chore: add .direnv/ to .gitignore to prevent accidental commits

* Expand unit tests for mcp-lambda-handler to maximize coverage (95%) and cover all error/edge cases

* Remove redundant test_lambda_server.py, move and rename comprehensive coverage test to tests/test_lambda_handler.py

* chore(build): precommit

* chore(build): remove local precommit

* chore(build): remove direnv

* chore(build): fix pyright

* chore(build): fix build

* fix: delete .envrc

---------

Signed-off-by: Scott Schreckengaust <scottschreckengaust@users.noreply.github.com>
Co-authored-by: Scott Schreckengaust <scottschreckengaust@users.noreply.github.com>
Co-authored-by: Laith Al-Saadoon <9553966+theagenticguy@users.noreply.github.com>
Co-authored-by: krokoko <alkrok@amazon.com>

awslabs-mcp committed:
Automatic update of packages

bx9900 committed:
Migrate over more tools from typscript package, add more tests, remove api-level tools

oferwix committed:
feat(terraform-mcp-server): add Terragrunt execution support (#256)

* feat(terraform-mcp-server): add Terragrunt execution support

Add functionality to execute Terragrunt commands alongside existing Terraform support.
This enables users to manage layered configurations with inheritance across environments.

Key additions:
- Implement ExecuteTerragruntCommand tool
- Add support for run-all and terragrunt-specific flags
- Include documentation and examples for Terragrunt workflows
- Update server.py to expose Terragrunt capabilities

* fix(terraform-mcp-server): remove unintended space in URL

* fix(terraform-mcp-server): remove used content

* fix(adding-terragrunt-support): remove un needed terragrunt mcp instructions

* fix: pre-commit

* fix: pyright errors

---------

Co-authored-by: Alain Krok <alkrok@amazon.com>
Co-authored-by: alexa-perlov <56009415+alexa-perlov@users.noreply.github.com>
Co-authored-by: Alexa Perlov <a.perlov33@gmail.com>
Co-authored-by: Scott Schreckengaust <scottschreckengaust@users.noreply.github.com>

mmouniro commented on src/aws-serverless-mcp-server/README.md:
```diff
@@ -0,0 +1,378 @@
+# AWS Serverless MCP Server
+
+A Model Context Protocol (MCP) server that provides tools and resources for deploying and managing applications on AWS Lambda.
```
suggestion : The main goal of the MCP server is to help build those applications . We do that by providing tools for deploying and managing them. 

question : should we mention serverless here ?

greyshi replied:
I think we should mention serverless, since applications will have more components than just Lambda functions

bx9900 replied:
yeah we should mention serverless, will update description.

mmouniro commented on src/aws-serverless-mcp-server/README.md:
```diff
@@ -0,0 +1,378 @@
+
+## Overview
+
+This MCP server enables AI assistants to deploy, update, invoke, and manage AWS Lambda functions and related serverless resources. It provides a set of tools and resources that can be used to interact with AWS Serverless services.
```
suggestion : also implements ?

mmouniro commented on src/aws-serverless-mcp-server/README.md:
```diff
@@ -0,0 +1,378 @@
+    "awslabs.aws-serverless-mcp": {
+      "command": "uvx",
+      "args": [
+        "awslabs.aws_serverless_mcp_server@latest"
```
question : All the MCP servers allow customers to configure their aws region/profile and fast mcp log level. Are we missing that feature ? If it's implemented let add it here in the example . 

Take a look at https://github.com/awslabs/mcp/tree/main/src/aws-location-mcp-server for inspiration

bx9900 replied:
I'll add support for this.

mmouniro commented on src/aws-serverless-mcp-server/README.md:
```diff
@@ -0,0 +1,378 @@
+
+The server provides the following resources:
+
+### Template Resources
```
question : Curious why those are resources and not tools ?

bx9900 replied:
I think the main difference is resources are client-managed, and tools are model-managed. As a resource, the AI client would decide when it needs to fetch a resource. Templates are files stored locally on the client machine, so thought it would make sense to have them as resources.

mmouniro commented on src/aws-serverless-mcp-server/README.md:
```diff
@@ -0,0 +1,378 @@
+
+Builds a serverless application using AWS SAM (Serverless Application Model) CLI.
+
+**Parameters:**
```
question : Do we need the full list of parameters in the README ? We do not expect humans to actually need those

greyshi replied:
While I agree, it does seem like listing all parameters in the readme is typical. e.g. https://github.com/github/github-mcp-server

mmouniro commented on src/aws-serverless-mcp-server/README.md:
```diff
@@ -0,0 +1,378 @@
+
+## Example Usage
+
+### Creating a Lambda Function with SAM
```
thought: I would have expected in the examples prompts that the human could use to leverage the MCP server

bx9900 replied:
Yeah this was LLM generated README. I'll go through and update examples.

greyshi replied:
Agreed. This almost seems like a readme for the AI assistant on how to use this mcp server rather than for the user. If that's standard practice that's fine, but I'm not sure it is.

mmouniro commented on src/aws-serverless-mcp-server/awslabs/aws_serverless_mcp_server/models.py:
```diff
@@ -0,0 +1,734 @@
+    PYTHON312 = "python3.12"
+    PYTHON313 = "python3.13"
+    JAVA17 = "java17"
+    JAVA21 = "java21"
```
question : this look like a subset of the lambda supported runtimes. is it on purpose ? 

https://docs.aws.amazon.com/lambda/latest/dg/lambda-runtimes.html#runtimes-supported

bx9900 replied:
Will remove this. We don't use this anywhere, and it'll be difficult to keep it updated as we release new runtimes.

mmouniro commented on src/aws-serverless-mcp-server/pyproject.toml:
```diff
@@ -0,0 +1,132 @@
+[project]
+name = "awslabs.aws-serverless-mcp-server"
```
chore: currently we are using serverless-mcp-server and lambda-mcp-server inconsistently. Let make all references to mcp server in the code be the former

mmouniro commented on src/aws-serverless-mcp-server/awslabs/aws_serverless_mcp_server/server.py:
```diff
@@ -0,0 +1,311 @@
+    Configure a custom domain for a deployed web application.
+    This tool updates a CloudFront distribution to use a custom domain and optionally creates a Route 53 record.
+    """)
+async def configure_domain_tool(
```
question : In the spirit of the discussion we had on Iac vs Manual should this also be done via the sam template instead of directly calling the API ? 

Should we replace this by guidance on how the LLM can update the the domain informations

bx9900 replied:
I copied this one over from Harold's MCP. I'm not sure if there are some CFN limitations that led us to create to use the APIs directly, but I'll try to make this part of the SAM template.

mmouniro commented on src/aws-serverless-mcp-server/awslabs/aws_serverless_mcp_server/server.py:
```diff
@@ -0,0 +1,311 @@
+@mcp.tool(description="""
+    Deploys a serverless application using AWS SAM (Serverless Application Model) CLI.
+    This command deploys your application to AWS CloudFormation.
+    Before running this tool, the application should already be built with 'sam_build' tool.
```
question : should we explicitly mention here that *each* time we do deploy we should have done build before ?

mmouniro commented on src/aws-serverless-mcp-server/awslabs/aws_serverless_mcp_server/server.py:
```diff
@@ -0,0 +1,311 @@
+    Builds a serverless application using AWS SAM (Serverless Application Model) CLI.
+    This command compiles your Lambda functions, creates deployment artifacts, and prepares your application for deployment.
+    Before running this tool, the application should already be initialized with 'sam_init' tool.
+    You should have AWS SAM CLI installed and configured in your environment.
```
question : Do we have guidance on how to install the SAM CLI ?

bx9900 replied:
I think in the README we can include links to SAM CLI installation. This way when the server is installed, the AI client will read the README and install dependencies as well. Having a tool to actually install SAM CLI would be too complex as we'd have to handle different operating systems.

mmouniro replied:
I actually think that we should provide guidance to the llm on how to do the install . It will likely be easier for the LLM to do the install than for the human to go read the documentation and do the install

mmouniro commented on src/aws-serverless-mcp-server/awslabs/aws_serverless_mcp_server/server.py:
```diff
@@ -0,0 +1,311 @@
+    return await handle_deployment_details(project_name)
+
+@mcp.tool(description="""
+    Builds a serverless application using AWS SAM (Serverless Application Model) CLI.
```
question : Should we provide the LLM more resources around SAM and SAM best practices . For instance https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/using-sam-cli-build.html

bx9900 replied:
Sure were you thinking just linking the docs or include them inline with description?

mmouniro commented on src/aws-serverless-mcp-server/awslabs/aws_serverless_mcp_server/models.py:
```diff
@@ -0,0 +1,734 @@
+from enum import Enum
+
+# SAM Models
+class SamInitRequest(BaseModel):
```
question: In the initial launch do we need to expose all SAM attributes are can we be more opinionated and expose only a subset of attribute that we think will be useful for experimentation/ vide coding ? 

For instance do we need to expose all the docker related attributes ?

bx9900 replied:
I don't think we need all options here, I'll reduce this down to a core set of options.

mmouniro commented on src/aws-serverless-mcp-server/awslabs/aws_serverless_mcp_server/tools/sam/sam_logs.py:
```diff
@@ -0,0 +1,102 @@
+    try:
+        function_name = request.function_name
+        stack_name = request.stack_name
+        tail = request.tail
```
question : Can we actually support tailing via the MCP server . I would expect that this will not work .

bx9900 replied:
MCP server on HTTP protocol can support streaming I think. With stdio, it probably will not work.

mmouniro commented on src/aws-serverless-mcp-server/awslabs/aws_serverless_mcp_server/tools/sam/sam_pipeline.py:
```diff
@@ -0,0 +1,203 @@
+from awslabs.aws_serverless_mcp_server.models import SamPipelineRequest
+from awslabs.aws_serverless_mcp_server.utils.logger import logger
+
+async def sam_pipeline(request: SamPipelineRequest) -> Dict[str, Any]:
```
chore: Seems like you removed the sam_pipeline tool but kept the underlying code .

mmouniro commented on src/aws-serverless-mcp-server/awslabs/aws_serverless_mcp_server/tools/sam/sam_logs.py:
```diff
@@ -0,0 +1,102 @@
+from awslabs.aws_serverless_mcp_server.models import SamLogsRequest
+from awslabs.aws_serverless_mcp_server.utils.logger import logger
+
+async def sam_logs(request: SamLogsRequest) -> Dict[str, Any]:
```
issue : Seems like there is not tools attached to sam_logs

mmouniro commented on src/aws-serverless-mcp-server/awslabs/aws_serverless_mcp_server/tools/sam/sam_build.py:
```diff
@@ -0,0 +1,100 @@
+    if request.base_dir:
+        cmd.extend(["--base-dir", request.base_dir])
+    if request.build_dir:
+        cmd.extend(["--build-dir", request.build_dir])
```
issue: Do we need to instruct the LLM to provide absolute path here and never related paths . If the LLM provides related path then we will be creating the projects relative to where the mcp server is running not where the customer project is.

bx9900 replied:
Will do. I think not just for this one but, for all parameters taking in paths they should be absolute.

mmouniro commented on src/aws-serverless-mcp-server/awslabs/aws_serverless_mcp_server/tools/sam/sam_deploy.py:
```diff
@@ -0,0 +1,90 @@
+    if request.config_env:
+        cmd.extend(["--config-env", request.config_env])
+    if request.guided_deploy:
+        cmd.append("--guided")
```
issue: guided will not work given that this is executing on the MCP server vs locally for on the customer cli directly

mmouniro commented on src/aws-serverless-mcp-server/awslabs/aws_serverless_mcp_server/tools/sam/sam_deploy.py:
```diff
@@ -0,0 +1,90 @@
+    if request.resolve_s3:
+        cmd.append("--resolve-s3")
+    if request.disable_rollback:
+        cmd.append("--disable-rollback")
```
issue : How the customer recover from a failure is they endup disabling rollbacks

mmouniro commented on src/aws-serverless-mcp-server/awslabs/aws_serverless_mcp_server/tools/webapps/get_logs.py:
```diff
@@ -0,0 +1,127 @@
+from awslabs.aws_serverless_mcp_server.models import GetLogsRequest
+from awslabs.aws_serverless_mcp_server.utils.logger import logger
+
+async def get_logs(request: GetLogsRequest) -> Dict[str, Any]:
```
question : Should we be using the sam logs instead as it automatically know how to get logs for resources beyond lambda ?

bx9900 replied:
Yes, we can remove get_logs tool.

mmouniro commented on src/aws-serverless-mcp-server/awslabs/aws_serverless_mcp_server/server.py:
```diff
@@ -0,0 +1,311 @@
+    return response
+
+@mcp.tool(description="""
+    Get logs from a deployed web application.
```
chore : We should expand the description here to give the LLM more guidances on when to use this tool. For instance something like `Use this tool during debugging of failing executions to get the logs of the execution and determine the root cause of the failure`

mmouniro commented on src/aws-serverless-mcp-server/awslabs/aws_serverless_mcp_server/server.py:
```diff
@@ -0,0 +1,311 @@
+    Get metrics from a deployed web application.
+    This tool retrieves CloudWatch metrics for a deployed application.
+    """)
+async def get_metrics_tool(
```
Question : How are we expecting the LLM to use the metrics it get from this tools ? When it's suppose to use it ?

bx9900 replied:
If user prompts for a dashboard of their application metrics, LLM can get raw data with this tool and build a visualization.

mmouniro commented on src/aws-serverless-mcp-server/awslabs/aws_serverless_mcp_server/server.py:
```diff
@@ -0,0 +1,311 @@
+    so that applications can be written in a standard web framework like Express or Next.js can be easily deployed to Lambda. You do not need to use
+    setup the code with any other adapter framework when using this tool.
+    """)
+async def deploy_webapp_tool(
```
question : Can we get "sam deploy" to do the same or it's a different use case ?

bx9900 replied:
This tool is focused towards deploying web applications using Lambda Web Adapter. It generates SAM templates using the bundled templates (frontend, fullstack, backend), whereas sam deploy tool is relying on the LLM to generate the template, or using the templates with sam init. It also can upload static assets to S3, which isn't supported by CFN/SAM.

mmouniro reviewed (commented):
Done have a review so far focused on the sam related tools. 
Overall feedback is that 

* The tools would gain to have a more expensive documentation that explain when the LLM should use them and what result they will help the LLM achieve
* Seems like some tools (deploy_webapps,get_logs) are duplicated from features that sam would provide . We should either explicitly explain why we can't use sam to achieve the same  or make them use sam 
* We are currently exposing all sam attribute even those that will not work because the execution of sam is done via the MCP server and not directly by the user. For instance something like guided on deploy will not work. More broadly I think we should expose only a subset of sam attribute that are useful in the context of the intent of this MCP server .

kennthhz committed:
feat(MySql): Add MCP support for Aurora MySQL (#389)

* Add MCP support for Aurora MySQL

* chore(build): precommit

---------

Co-authored-by: Alain Krok <alkrok@amazon.com>

awslabs-mcp committed:
Automatic update of packages

greyshi commented on src/aws-serverless-mcp-server/README.md:
```diff
@@ -0,0 +1,378 @@
+
+A Model Context Protocol (MCP) server that provides tools and resources for deploying and managing applications on AWS Lambda.
+
+## Overview
```
Once we add the new "enable-write" and "enable-sensitive-data" flags, these should be prominently highlighted in the readme.

greyshi commented on src/aws-serverless-mcp-server/awslabs/aws_serverless_mcp_server/server.py:
```diff
@@ -0,0 +1,311 @@
+# Template resources
+@mcp.resource("template://list")
+def template_list() -> Dict[str, Any]:
+    """List of available deployment templates."""
```
Can we be a bit more descriptive here to make it clear to the AI when to actually use this? What kind of deployments and what are the templates accomplishing?

greyshi commented on src/aws-serverless-mcp-server/awslabs/aws_serverless_mcp_server/server.py:
```diff
@@ -0,0 +1,311 @@
+
+@mcp.resource("template://{template_name}")
+def template_details(template_name: str) -> Dict[str, Any]:
+    """Deatils of a deployment template."""
```
(nit) typo

greyshi commented on src/aws-serverless-mcp-server/awslabs/aws_serverless_mcp_server/server.py:
```diff
@@ -0,0 +1,311 @@
+    return response
+
+@mcp.tool(description="""
+    Deploy web applications to AWS, including database resources like DynamoDB tables. This tool uses the Lambda Web Adapter framework
```
Just listing DynamoDB makes could confuse the AI to think it's only applicable for that. Can we list all the resources that can be deployed using this?

greyshi commented on src/aws-serverless-mcp-server/awslabs/aws_serverless_mcp_server/server.py:
```diff
@@ -0,0 +1,311 @@
+    return response
+
+@mcp.tool(description="""
+    Get help information about web application deployments or deployment status.
```
Only web applications?

greyshi commented on src/aws-serverless-mcp-server/awslabs/aws_serverless_mcp_server/template/templates/README.md:
```diff
@@ -0,0 +1,38 @@
+# Deployment Templates
+
+This directory contains Handlebars templates for different types of deployments.
```
Should we describe what Handlebars templates are?

greyshi commented on src/aws-serverless-mcp-server/awslabs/aws_serverless_mcp_server/tools/sam/sam_init.py:
```diff
@@ -0,0 +1,108 @@
+
+async def sam_init(request: SamInitRequest) -> Dict[str, Any]:
+    """
+    Initialize a serverless application with an AWS SAM template
```
Could be worth including: What specifically does initializing a serverless app entail? What's usually defined in the SAM template?

greyshi commented on src/aws-serverless-mcp-server/awslabs/aws_serverless_mcp_server/tools/webapps/utils/dependency_installer.py:
```diff
@@ -0,0 +1,212 @@
+
+"""Dependency Installer for AWS Serverless MCP Server.
+
+Handles installation of dependencies in the build artifacts directory
```
Can we include which runtimes are supported?

greyshi commented on src/aws-serverless-mcp-server/awslabs/aws_serverless_mcp_server/tools/webapps/utils/startup_script_generator.py:
```diff
@@ -0,0 +1,178 @@
+    additional_env: Optional[Dict[str, str]] = None
+) -> str:
+    """
+    Generate a startup script based on runtime and entry point.
```
Can we add what purpose a startup script serves?

mikegc-aws committed:
fix(lambda-handler): Fix the status code for notifications (#394)

awslabs-mcp committed:
Automatic update of packages

krokoko committed:
chore(doc): update readme (#391)

* chore(readme): remove duplicate

* chore(readme): remove duplicate

* chore(readme): add missing server to doc

* chore(readme): add missing server to doc

* chore(readme): precommit

krokoko committed:
chore(doc): update readmes (#405)

* chore(template): update issue template

* chore(doc): correct name services

* chore(doc): correct readmes

* chore(doc): update readmes

krokoko committed:
chore(doc): improve service naming consistency and fix links (#407)

* chore(readme): correct readme

Benjscho committed:
perf: Dsql connection reuse (#400)

* Reuse DSQL connections

This adds reuse to DSQL connections. The DSQL MCP server now maintains a
persistent connection and reuses it across queries.

* remove unused consts

* Fix dsql unit tests

* add session token to docker ex

* Add --profile option to dsql

Add a --profile option as well as AWS_PROFILE.

* DSQL readme updates

* Don't validate connection

This saves a round trip on execution by just trying to use the
connection and creating a new one if it fails before retrying execution.

* fix pyright

* Add coverage

---------

Co-authored-by: Alain Krok <alkrok@amazon.com>

awslabs-mcp committed:
Automatic update of packages

tomron-aws committed:
Update README.md (#409)

Updated ReadMe installation instructions to use remote server address for mcp server config

bx9900 committed:
Updates to readme, tool descriptions

erbenmo committed:
fix: support read-only mode for dynamodb mcp (#414)

* implement list import

* add read-only flag

dependabot[bot] committed:
chore(deps): update docker images: bump the docker-version-updates group across 21 directories with 2 updates (#415)

Bumps the docker-version-updates group with 1 update in the /src/amazon-kendra-index-mcp-server directory: sam/build-python3.10.
Bumps the docker-version-updates group with 1 update in the /src/amazon-mq-mcp-server directory: sam/build-python3.10.
Bumps the docker-version-updates group with 1 update in the /src/amazon-neptune-mcp-server directory: sam/build-python3.10.
Bumps the docker-version-updates group with 1 update in the /src/amazon-sns-sqs-mcp-server directory: sam/build-python3.10.
Bumps the docker-version-updates group with 1 update in the /src/aurora-dsql-mcp-server directory: sam/build-python3.10.
Bumps the docker-version-updates group with 1 update in the /src/aws-diagram-mcp-server directory: sam/build-python3.10.
Bumps the docker-version-updates group with 1 update in the /src/aws-documentation-mcp-server directory: sam/build-python3.13.
Bumps the docker-version-updates group with 1 update in the /src/aws-location-mcp-server directory: sam/build-python3.10.
Bumps the docker-version-updates group with 1 update in the /src/bedrock-kb-retrieval-mcp-server directory: sam/build-python3.10.
Bumps the docker-version-updates group with 1 update in the /src/cdk-mcp-server directory: sam/build-python3.10.
Bumps the docker-version-updates group with 1 update in the /src/cfn-mcp-server directory: sam/build-python3.10.
Bumps the docker-version-updates group with 1 update in the /src/core-mcp-server directory: sam/build-python3.10.
Bumps the docker-version-updates group with 1 update in the /src/cost-analysis-mcp-server directory: sam/build-python3.10.
Bumps the docker-version-updates group with 1 update in the /src/dynamodb-mcp-server directory: sam/build-python3.10.
Bumps the docker-version-updates group with 1 update in the /src/lambda-mcp-server directory: sam/build-python3.10.
Bumps the docker-version-updates group with 1 update in the /src/memcached-mcp-server directory: sam/build-python3.10.
Bumps the docker-version-updates group with 1 update in the /src/mysql-mcp-server directory: sam/build-python3.10.
Bumps the docker-version-updates group with 1 update in the /src/nova-canvas-mcp-server directory: sam/build-python3.10.
Bumps the docker-version-updates group with 1 update in the /src/postgres-mcp-server directory: sam/build-python3.10.
Bumps the docker-version-updates group with 1 update in the /src/terraform-mcp-server directory: sam/build-python3.10.
Bumps the docker-version-updates group with 1 update in the /src/valkey-mcp-server directory: sam/build-python3.10.


Updates `sam/build-python3.10` from `04cdbe8` to `e78695d`

Updates `sam/build-python3.10` from `a40f492` to `e78695d`

Updates `sam/build-python3.10` from `a40f492` to `e78695d`

Updates `sam/build-python3.10` from `a40f492` to `e78695d`

Updates `sam/build-python3.10` from `a40f492` to `e78695d`

Updates `sam/build-python3.10` from `a40f492` to `e78695d`

Updates `sam/build-python3.13` from `13645e3` to `387fc3f`

Updates `sam/build-python3.10` from `a40f492` to `e78695d`

Updates `sam/build-python3.10` from `a40f492` to `e78695d`

Updates `sam/build-python3.10` from `a40f492` to `e78695d`

Updates `sam/build-python3.10` from `a40f492` to `e78695d`

Updates `sam/build-python3.10` from `a40f492` to `e78695d`

Updates `sam/build-python3.10` from `a40f492` to `e78695d`

Updates `sam/build-python3.10` from `a40f492` to `e78695d`

Updates `sam/build-python3.10` from `a40f492` to `e78695d`

Updates `sam/build-python3.10` from `a40f492` to `e78695d`

Updates `sam/build-python3.10` from `04cdbe8` to `e78695d`

Updates `sam/build-python3.10` from `a40f492` to `e78695d`

Updates `sam/build-python3.10` from `a40f492` to `e78695d`

Updates `sam/build-python3.10` from `a40f492` to `e78695d`

Updates `sam/build-python3.10` from `a40f492` to `e78695d`

---
updated-dependencies:
- dependency-name: sam/build-python3.10
  dependency-version: e78695db10ca8cb129e59e30f7dc9789b0dbd0181dba195d68419c72bac51ac1
  dependency-type: direct:production
  dependency-group: docker-version-updates
- dependency-name: sam/build-python3.10
  dependency-version: e78695db10ca8cb129e59e30f7dc9789b0dbd0181dba195d68419c72bac51ac1
  dependency-type: direct:production
  dependency-group: docker-version-updates
- dependency-name: sam/build-python3.10
  dependency-version: e78695db10ca8cb129e59e30f7dc9789b0dbd0181dba195d68419c72bac51ac1
  dependency-type: direct:production
  dependency-group: docker-version-updates
- dependency-name: sam/build-python3.10
  dependency-version: e78695db10ca8cb129e59e30f7dc9789b0dbd0181dba195d68419c72bac51ac1
  dependency-type: direct:production
  dependency-group: docker-version-updates
- dependency-name: sam/build-python3.10
  dependency-version: e78695db10ca8cb129e59e30f7dc9789b0dbd0181dba195d68419c72bac51ac1
  dependency-type: direct:production
  dependency-group: docker-version-updates
- dependency-name: sam/build-python3.10
  dependency-version: e78695db10ca8cb129e59e30f7dc9789b0dbd0181dba195d68419c72bac51ac1
  dependency-type: direct:production
  dependency-group: docker-version-updates
- dependency-name: sam/build-python3.13
  dependency-version: 387fc3f8ee99bd56267bc52513e6669074b2d197dff273e03b639b6afca51790
  dependency-type: direct:production
  dependency-group: docker-version-updates
- dependency-name: sam/build-python3.10
  dependency-version: e78695db10ca8cb129e59e30f7dc9789b0dbd0181dba195d68419c72bac51ac1
  dependency-type: direct:production
  dependency-group: docker-version-updates
- dependency-name: sam/build-python3.10
  dependency-version: e78695db10ca8cb129e59e30f7dc9789b0dbd0181dba195d68419c72bac51ac1
  dependency-type: direct:production
  dependency-group: docker-version-updates
- dependency-name: sam/build-python3.10
  dependency-version: e78695db10ca8cb129e59e30f7dc9789b0dbd0181dba195d68419c72bac51ac1
  dependency-type: direct:production
  dependency-group: docker-version-updates
- dependency-name: sam/build-python3.10
  dependency-version: e78695db10ca8cb129e59e30f7dc9789b0dbd0181dba195d68419c72bac51ac1
  dependency-type: direct:production
  dependency-group: docker-version-updates
- dependency-name: sam/build-python3.10
  dependency-version: e78695db10ca8cb129e59e30f7dc9789b0dbd0181dba195d68419c72bac51ac1
  dependency-type: direct:production
  dependency-group: docker-version-updates
- dependency-name: sam/build-python3.10
  dependency-version: e78695db10ca8cb129e59e30f7dc9789b0dbd0181dba195d68419c72bac51ac1
  dependency-type: direct:production
  dependency-group: docker-version-updates
- dependency-name: sam/build-python3.10
  dependency-version: e78695db10ca8cb129e59e30f7dc9789b0dbd0181dba195d68419c72bac51ac1
  dependency-type: direct:production
  dependency-group: docker-version-updates
- dependency-name: sam/build-python3.10
  dependency-version: e78695db10ca8cb129e59e30f7dc9789b0dbd0181dba195d68419c72bac51ac1
  dependency-type: direct:production
  dependency-group: docker-version-updates
- dependency-name: sam/build-python3.10
  dependency-version: e78695db10ca8cb129e59e30f7dc9789b0dbd0181dba195d68419c72bac51ac1
  dependency-type: direct:production
  dependency-group: docker-version-updates
- dependency-name: sam/build-python3.10
  dependency-version: e78695db10ca8cb129e59e30f7dc9789b0dbd0181dba195d68419c72bac51ac1
  dependency-type: direct:production
  dependency-group: docker-version-updates
- dependency-name: sam/build-python3.10
  dependency-version: e78695db10ca8cb129e59e30f7dc9789b0dbd0181dba195d68419c72bac51ac1
  dependency-type: direct:production
  dependency-group: docker-version-updates
- dependency-name: sam/build-python3.10
  dependency-version: e78695db10ca8cb129e59e30f7dc9789b0dbd0181dba195d68419c72bac51ac1
  dependency-type: direct:production
  dependency-group: docker-version-updates
- dependency-name: sam/build-python3.10
  dependency-version: e78695db10ca8cb129e59e30f7dc9789b0dbd0181dba195d68419c72bac51ac1
  dependency-type: direct:production
  dependency-group: docker-version-updates
- dependency-name: sam/build-python3.10
  dependency-version: e78695db10ca8cb129e59e30f7dc9789b0dbd0181dba195d68419c72bac51ac1
  dependency-type: direct:production
  dependency-group: docker-version-updates
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

bilalshirazi committed:
fix(doc): JSON syntax errors in mcp.json example (#412)

Co-authored-by: Scott Schreckengaust <scottschreckengaust@users.noreply.github.com>

dependabot[bot] committed:
chore(deps): update github-actions: bump the github-actions-version-updates group with 2 updates (#416)

Bumps the github-actions-version-updates group with 2 updates: [bridgecrewio/checkov-action](https://github.com/bridgecrewio/checkov-action) and [astral-sh/setup-uv](https://github.com/astral-sh/setup-uv).


Updates `bridgecrewio/checkov-action` from 12.3014.0 to 12.3022.0
- [Release notes](https://github.com/bridgecrewio/checkov-action/releases)
- [Commits](https://github.com/bridgecrewio/checkov-action/compare/360818f2ad44468d3294cfddae854a8c9036dfee...ff7aee4d041716dcce856c7bd319cb7048cb37e7)

Updates `astral-sh/setup-uv` from 6.0.1 to 6.1.0
- [Release notes](https://github.com/astral-sh/setup-uv/releases)
- [Commits](https://github.com/astral-sh/setup-uv/compare/6b9c6063abd6010835644d4c2e1bef4cf5cd0fca...f0ec1fc3b38f5e7cd731bb6ce540c5af426746bb)

---
updated-dependencies:
- dependency-name: bridgecrewio/checkov-action
  dependency-version: 12.3022.0
  dependency-type: direct:production
  update-type: version-update:semver-minor
  dependency-group: github-actions-version-updates
- dependency-name: astral-sh/setup-uv
  dependency-version: 6.1.0
  dependency-type: direct:production
  update-type: version-update:semver-minor
  dependency-group: github-actions-version-updates
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

awslabs-mcp committed:
Automatic update of packages

kenliao94 committed:
remove SSE support (#417)

theagenticguy committed:
fix(kendra-mcp)!: remove sse (#419)

* fix(kendra-mcp): remove sse

* fix: commitizen version pyproject

* fix: update patch so bumps to zero

---------

Co-authored-by: Scott Schreckengaust <scottschreckengaust@users.noreply.github.com>

PaulVincent707 committed:
fix!: remove SSE support from amazon-sns-sqs-mcp-server (#418)

* fix: remove SSE support from amazon-sns-sqs-mcp-server

* fix: precommit errors

* fix: major version bump

* fix: commitizen version pyproject

* fix: update patch so bumps to zero

---------

Co-authored-by: laithalsaadoon <alsaadoonlaith@gmail.com>
Co-authored-by: Scott Schreckengaust <scottschreckengaust@users.noreply.github.com>

bx9900 committed:
Remove unused parameters from tools input models

bx9900 committed:
Add new security flags and clean up code

awslabs-mcp committed:
Automatic update of packages

theagenticguy committed:
fix: remove sse from multiple mcp servers (#421)

* fix: remove sse from neptune mcp server

* fix: remove sse test from neptune

* fix: remove sse dsql mcp server

* fix: remove sse - aws-diagram-mcp-server

* fix: remove sse aws documentation mcp server

* fix: remove sse - aws-location-mcp-server

* fix: remove sse bedrock kb server

* fix: remove sse - aws-location-mcp-server

* fix: location test fixes

* fix: remove sse - code-doc-gen-mcp-server

* fix: remove sse cfn server

* fix: remove sse core server

* fix: remove sse documentdb server

* fix: remove sse - code-doc-gen-mcp-server

* fix: remove sse - cost-analysis-mcp-server

* fix: remove sse - dynamodb-mcp-server

* fix: remove sse - git-repo-research-mcp-server

* fix: remove sse - mysql-mcp-server

* fix: remove sse - postgres-mcp-server

* fix: remove sse - terraform-mcp-server

* fix: remove sse - terraform-mcp-server

* fix: remove sse - terraform-mcp-server

* fix: remove sse frontend server

* fix: remove sse - terraform-mcp-server

* fix: remove sse lambda_mcp_server

* fix: remove sse - terraform-mcp-server

* fix: remove sse memcached

* fix: remove sse nova

* fix: remove sse synthetic

* fix: remove sse valkey

* fix: woops, fixed synthetic again

* fix: terraform tests

* fix: ensure all packages are ready for major bump

* fix: add SSE support removal notice to README and MCP server changelogs

- Add Server Sent Events support removal notice to root README
- Document SSE removal in major version changelogs for all MCP servers
- Reference MCP specification backwards compatibility guidelines
- Mention future Streamable HTTP support plans
- Applies to major versions released on 2025-05-26

* fix: remove magicmock from lambda mcp server

* fix: remove magicmock from lambda mcp server

* fix: remove sse - amazon-mq-mcp-server

* fix: undo some changelogs and test removals

---------

Co-authored-by: Paul Vincent <pauvince@amazon.com>

awslabs-mcp committed:
Automatic update of packages

bx9900 committed:
Update README with security controls details

bx9900 committed:
Update configure_domain tool to create certificates and remove unused models.

bx9900 committed:
Add CODEOWNERS for serverless mcp

mmouniro commented on src/aws-serverless-mcp-server/README.md:
```diff
@@ -0,0 +1,601 @@
+This Model Context Protocol (MCP) server enables AI assistants to build and deploy applications onto AWS Serverless using Serverless Application Model (SAM). It implements a set of tools and resources that can be used to interact with serverless services.
+
+## Features
+This MCP server acts as a bridge between MCP clients and AWS Serverless services, allowing generative AI models to create, configure, and manage Serverless resources.
```
question : Is that really what we want to do ? The intent of this MCP is not to manage serveless resources it's to help customers write code for serverless services . The fact that we can deploy/create those resources is a side effect of needing to test the code

mmouniro commented on src/aws-serverless-mcp-server/README.md:
````diff
@@ -0,0 +1,601 @@
+Example user prompt:
+
+```
+I need to deploy a Node.js backend web application to AWS Lambda. The project is called "my-web-app" and is located at "/path/to/project". The built artifacts are in "/path/to/artifacts". The app runs on port 3000 and needs 512MB of memory with a 30-second timeout. Can you deploy this for me?
````
question : Is the creating of the webapp also part of this MCP server ? If yes let add it to the example prompt

mmouniro commented on src/aws-serverless-mcp-server/README.md:
````diff
@@ -0,0 +1,601 @@
+Example user prompt:
+
+```
+I need to build my SAM application located in "/projects/aws-lambda/weather-api". Can you build it for me? I'd like to use a container for the build process and cache the build artifacts for faster builds in the future.
````
question : What would happen if I do not specify things like container for build process and caching ? Should those be baked directly into the description of the tools ?

mmouniro commented on src/aws-serverless-mcp-server/README.md:
````diff
@@ -0,0 +1,601 @@
+Example user prompt:
+
+```
+I want to test my Lambda function locally before deploying it. Can you invoke the "GetWeatherFunction" from my SAM project in "/projects/aws-lambda/weather-api"? Here's the test event data: {"city": "Seattle", "country": "USA"}
````
thought : broadly I have the example that the example are very specific . I would not expect a casual usual onboarding to the service to have this level of detail . Should we be extracting the instruction to the tools descriptions ? 

For instance I would be able to say `Can you test the application now` and the LLM should ask me what is the test event that I want to use . Better it can generate it for me automatically or guide me to add a schema to by function to get better validation.

mmouniro commented on src/aws-serverless-mcp-server/README.md:
````diff
@@ -0,0 +1,601 @@
+Example user prompt:
+
+```
+I've deployed my web application "my-portfolio" to AWS and now I want to use my custom domain "example.com" for it. I already have an ACM certificate with ARN "arn:aws:acm:us-east-1:123456789012:certificate/abcd1234" and my Route 53 hosted zone ID is "Z1234ABCD5678EF". Can you help me set this up?
````
question : Is it a realistic scenario for a vide coder ? I would expect that they have no prior resource and it will be more like `can you provide me the endpoint to test the application ?`

mmouniro commented on src/aws-serverless-mcp-server/README.md:
```diff
@@ -0,0 +1,601 @@
+### Production Use Cases
+The AWS Serverless MCP Server can be used for production environments with proper security controls in place. For production use cases, consider the following:
+
+* **Read-Only Mode by Default**: The server runs in read-only mode by default, which is safer for production environments. Only explicitly enable write access when necessary.
```
Should we recommend folks to use this for production use ? This is meant for vibe coding I think we should make it very explicit.

alexa-perlov committed:
fix(samples): Update MultiServerMCPClient usage for langchain-mcp-adapters 0.1.0 (#410)

* fix: Update MultiServerMCPClient usage for langchain-mcp-adapters 0.1.0

* fix: Update MultiServerMCPClient usage for langchain-mcp-adapters 0.1.0 for kb sample

* chore: cleaning up repetitive logs + vars

bx9900 committed:
Let boto3 auto-detect region if not provided

bx9900 committed:
Remove route 53 record set creation from deployWebAppTool

scottschreckengaust committed:
Merge branch 'main' into serverless-mcp

scottschreckengaust committed:
feat: anthropic model with variable (#426)

* feat: control anthropic model with variable

* fix: missing curly bracket

Commit:
fix(tools): rename search_research_repository_suggestions to search_repositories_on_github (#397)

* fix(tools): rename search_research_repository_suggestions to search_repositories_on_github

Rename tool from search_research_repository_suggestions to search_repositories_on_github
to resolve name length issue when combined with server name in MCP client integration.

The previous tool name exceeded the 64 character limit when prefixed with the server name,
causing the Git Repo Research MCP Server to fail loading in MCP clients.

Fixes #361

* fix(tools): rename search_research_repository_suggestions to search_repositories_on_github

Rename tool from search_research_repository_suggestions to search_repositories_on_github
to resolve name length issue when combined with server name in MCP client integration.

The previous tool name exceeded the 64 character limit when prefixed with the server name,
causing the Git Repo Research MCP Server to fail loading in MCP clients.

Fixes #361

---------

Co-authored-by: Alain Krok <alkrok@amazon.com>
Co-authored-by: Laith Al-Saadoon <9553966+theagenticguy@users.noreply.github.com>
Co-authored-by: Scott Schreckengaust <scottschreckengaust@users.noreply.github.com>

bx9900 committed:
Update README

bx9900 committed:
Migrate from pybars to jinja

bx9900 committed:
Update serverless codeowners

scottschreckengaust committed:
fix: update CODEOWNERS (#383)

* fix: update CODEOWNERS

* fix: CODEOWNERS

* Update CODEOWNERS

* Update CODEOWNERS

* fix: names for cfn CODEOWNERS

---------

Co-authored-by: Alain Krok <alkrok@amazon.com>

bx9900 committed:
Update unit tests and add docs page

awslabs-mcp committed:
Automatic update of packages

joe-king-sh committed:
fix: multiple typos in documentation and error messages (#423)

* fix: correct typo in error message (occured → occurred)

* fix: correct typo in README files (succesful → successful)

* fix: correct typo in README files (ficticious → fictitious)

* fix: correct typo in README file (prinicpal → principal)

* fix: correct typo in README file (priveledge → privilege)

* fix: correct typo in README file (defualt → default)

---------

Co-authored-by: Scott Schreckengaust <scottschreckengaust@users.noreply.github.com>

bx9900 committed:
Fix pre-commit errors

bx9900 committed:
Fix bandit errors

bx9900 committed:
Create uv.lock file

bx9900 committed:
Fix unit tests

mmouniro committed:
feat(aws-serverless-mcp-server) : add support for the schema registry

bx9900 committed:
Mock github calls in serverless template test

bx9900 committed:
Fix pre-commit issues

gautric committed:
Cleanup Sample + Remove Duplicate link into Documentation section (#406)

Small fixes changes :

Cleanup JSON sample
Remove Duplicate link into Documentation section

Co-authored-by: Alain Krok <alkrok@amazon.com>
Co-authored-by: Scott Schreckengaust <scottschreckengaust@users.noreply.github.com>

bx9900 committed:
Fix pyright errors

bx9900 committed:
Fix merge conflict with README

bx9900 committed:
Merge branch 'main' into serverless-mcp

bx9900 committed:
Update tests after pyright changes

bx9900 committed:
Remove test for missing param since pyright validates already

bx9900 committed:
Fix pre-commit issues

bx9900 committed:
Add more unit tests for utils

KaramVSingh committed:
chore: Adding user agent to cfn mcp server (#434)

Co-authored-by: Karam Singh <amzkaram@amazon.com>

bx9900 committed:
Fix pre-commit issues

bx9900 committed:
Fix pre-commit

bx9900 committed:
update docs related to handle bar to jinja conversion

bx9900 committed:
Add test for update frontend tool

mmouniro committed:
feat(server): AWS Step Functions MCP Server (#298) (#377)

* chore : clone lambda mcp server

The step functions mcp server will build upon the pattern established by the lambda mcp server.
This initial change aim to have a baseline we can work with by cloning the existing lambda mcp server implementation.

* chore(stepfunctions-mcp-server): rename Lambda references to Step Functions

This commit updates all Lambda references to use Step Functions terminology without changing any logic.

* feat(stepfunctions-mcp-server): implement Step Functions API integration

This commit updates the server to use Step Functions APIs and adds support for both Standard and Express state machines.

* test(stepfunctions-mcp-server) : Add tests for the step functions integration

* chore(stepfunctions-tool-mcp-server) rename `AWS Step Functions MCP Server` to `AWS Step Functions Tool MCP Server`

This commit renames the AWS Step Functions MCP Server to AWS Step Functions Tool MCP Server to better reflect its purpose as a tool provider.

* chore(stepfunctions-tool-mcp-server) Move examples to the top level samples folder

* chore (step-functions-tool-mcp-server) : Remove sse support

* chore(stepfunctions-tool-mcp-server) : update codeowners

---------

Co-authored-by: Alain Krok <alkrok@amazon.com>
Co-authored-by: Scott Schreckengaust <scottschreckengaust@users.noreply.github.com>
Co-authored-by: Laith Al-Saadoon <9553966+theagenticguy@users.noreply.github.com>

bx9900 committed:
addres pre-commit errors

bx9900 committed:
Merge branch 'main' into serverless-mcp

awslabs-mcp committed:
Automatic update of packages

bx9900 committed:
Include examples in event schem input parmeter description

bx9900 committed:
Disable new line stripping

bx9900 committed:
Add pre-commit file and increase test coverage for server

bx9900 committed:
Fix pre-comit issues

bx9900 committed:
Fix bandit isssues

Shubhranshu153 committed:
feat: Implement finch mcp server (#408)

* feat: finch mcp server

Signed-off-by: Shubharanshu Mahapatra <shubhum@amazon.com>

* fix: update logic for image tags to be unique

Signed-off-by: Shubharanshu Mahapatra <shubhum@amazon.com>

* feat: add ecr create tool

Signed-off-by: Shubharanshu Mahapatra <shubhum@amazon.com>

* fix: cross platform support

Signed-off-by: Shubharanshu Mahapatra <shubhum@amazon.com>

* fix: remove stdout/stderr from tool response

Signed-off-by: Shubharanshu Mahapatra <shubhum@amazon.com>

* fix: logging of invocations in finch mcp server

Signed-off-by: Shubharanshu Mahapatra <shubhum@amazon.com>

* feat: add pytest to finch mcp server

Signed-off-by: Shubharanshu Mahapatra <shubhum@amazon.com>

* fix: Update documentation

Signed-off-by: Shubharanshu Mahapatra <shubhum@amazon.com>

* fix: switch logging to loguru

Signed-off-by: Shubharanshu Mahapatra <shubhum@amazon.com>

* fix: testing and logging

Signed-off-by: Shubharanshu Mahapatra <shubhum@amazon.com>

* fix: restrict build options and readme fixes

Signed-off-by: Shubharanshu Mahapatra <shubhum@amazon.com>

* fix: remove dockerfile for finch server as we dont support container in container builds

Signed-off-by: Shubharanshu Mahapatra <shubhum@amazon.com>

* fix: Refactor test and update logs

Signed-off-by: Shubharanshu Mahapatra <shubhum@amazon.com>

* fix: server tests and docs

Signed-off-by: Shubharanshu Mahapatra <shubhum@amazon.com>

* fix: result from check_ecr didn't contain changed bool

Signed-off-by: Shubharanshu Mahapatra <shubhum@amazon.com>

* feat: add readonly mode for AWS resource creation

Signed-off-by: Shubharanshu Mahapatra <shubhum@amazon.com>

* fix: update get_hash function to return short_hash

Signed-off-by: Shubharanshu Mahapatra <shubhum@amazon.com>

* fix: test mock for readonly mode

Signed-off-by: Shubharanshu Mahapatra <shubhum@amazon.com>

* fix: minor doc updates

Signed-off-by: Shubharanshu Mahapatra <shubhum@amazon.com>

* fix: update ecr and region regex pattern

Signed-off-by: Shubharanshu Mahapatra <shubhum@amazon.com>

* fix: change app_name to repository_name

Signed-off-by: Shubharanshu Mahapatra <shubhum@amazon.com>

* fix: add failure message on sensitive data filter

Signed-off-by: Shubharanshu Mahapatra <shubhum@amazon.com>

* fix: allow only finch command to be executed as part of execute_command

Signed-off-by: Shubharanshu Mahapatra <shubhum@amazon.com>

* fix: update error response for vm and path compatibility"

Signed-off-by: Shubharanshu Mahapatra <shubhum@amazon.com>

* docs: Add codeowners for finch

Signed-off-by: Shubharanshu Mahapatra <shubhum@amazon.com>

* docs: add uvx format in instruction and update version numbers

Signed-off-by: Shubharanshu Mahapatra <shubhum@amazon.com>

* fix: add additional checks for security patterns in subprocess run

Signed-off-by: Shubharanshu Mahapatra <shubhum@amazon.com>

* fix: minor updates to test and docs

Signed-off-by: Shubharanshu Mahapatra <shubhum@amazon.com>

* fix: change from using nested pydantic model to direct

Signed-off-by: Shubharanshu Mahapatra <shubhum@amazon.com>

---------

Signed-off-by: Shubharanshu Mahapatra <shubhum@amazon.com>

awslabs-mcp committed:
Automatic update of packages

bx9900 committed:
Flatten input parameters

bx9900 committed:
Merge branch 'main' into serverless-mcp

bx9900 committed:
Update README

bx9900 committed:
Undo changes from README.md

bx9900 committed:
Revert README back to main

patrick-yu-amzn committed:
feat: Amazon EKS MCP Server (#429)

* feat: bootstrap eks-mcp-server from template

* feat: eks mcp server full implementation

* fix: proper mocking of aws clients in tests

Some unit tests doesn't mock aws client creation properly and ends up
trying to create real clients, which doesn't work when the system
doesn't have aws configuration.

* fix: address all pyright errors

* fix: suppress checkov errors

* fix: more fixes copied from internal repo

* fix: README.md and tool doc update + CODEOWNERS

* fix: more doc fixes

* fix: add user-agent to boto client setup

This allows us to track eks-mcp-server tool usage.

---------

Co-authored-by: Scott Schreckengaust <scottschreckengaust@users.noreply.github.com>

bx9900 committed:
Fix README

bx9900 committed:
Fix pre-commmit

bx9900 committed:
Merge branch 'main' into serverless-mcp

awslabs-mcp committed:
Automatic update of packages

Shubhranshu153 committed:
docs: add finch docs to server docs dir (#438)

Signed-off-by: Shubharanshu Mahapatra <shubhum@amazon.com>

scottschreckengaust committed:
Merge branch 'main' into serverless-mcp

bx9900 committed:
Add user agent to boto3 calls

bx9900 committed:
Fix build

bx9900 committed:
Audit secrets

bx9900 committed:
Remove imports from __init__.py

bx9900 committed:
Match init file

bx9900 committed:
Address claude feedback

bx9900 committed:
remove unused file

hashimsharkh committed:
fix: Fix Q related Issues (#444)

* Fix Q issues

* Fix tests

* Fix tests

scottschreckengaust committed:
Merge branch 'main' into serverless-mcp

bx9900 committed:
Set mimetype for s3 upload and return stderr logs for local invoke

scottschreckengaust committed:
fix: package installation in README

awslabs-mcp committed:
Automatic update of packages

theagenticguy committed:
Merge branch 'main' into serverless-mcp

bx9900 committed:
Add serverless mcp to top level readme
